### PR TITLE
fix: fix name with `

### PR DESCRIPTION
**Pull request by JackTan25**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR
1. when we use update * or insert *, we will find the field in source by name, if user use \`name\`, it will can't find out, so we use alternate to do a special process.
2. fix a delete bug, need to send rowIds to downstream
- Closes #issue

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13163)

### CODE REVIEW

File: src/query/sql/src/planner/binder/merge_into.rs [hunk 1]
```diff
@@ -144,7 +144,7 @@ impl Binder {
             let mut name_map = HashMap::<String, Vec<IndexType>>::new();
             for column in source_output_columns {
                 name_map
-                    .entry(column.column_name.clone())
+                    .entry(column.column_name.as_str().trim_matches('`').to_string())
```
Comment (b41sh) [inline src/query/sql/src/planner/binder/merge_into.rs:147]: We should remove '`' from `column_name` when binding columns, `subquery` will also report errors due to this problem
```sql
mysql> select a from (select a from test);
+--------------------+
| a                  |
+--------------------+
| 0.0941881516339449 |
| 0.8833429252655464 |
+--------------------+
2 rows in set (0.05 sec)
Read 2 rows, 17.00 B in 0.007 sec., 306.9 rows/sec., 2.55 KiB/sec.

mysql> select a from (select `a` from test);
ERROR 1105 (HY000): SemanticError. Code: 1065, Text = error:
  --> SQL:1:8
  |
1 | select a from (select `a` from test)
  |        ^ column a doesn't exist

mysql> select `a` from (select `a` from test);
ERROR 1105 (HY000): SemanticError. Code: 1065, Text = error:
  --> SQL:1:8
  |
1 | select `a` from (select `a` from test)
  |        ^^^ column a doesn't exist, do you mean 'a'?
```

**Comment by JackTan25** [comment]
we should fix in resolve_column_name

**Comment by andylokandy** [comment]
I guess the problem occurs here (L266):
https://github.com/datafuselabs/databend/blob/609e02bf70680e319b0ebedf3c19f63ac3f85d8d/src/query/sql/src/planner/binder/project.rs#L263-L276

**Comment by JackTan25** [comment]
> I guess the problem occurs here (L266):
> 
> https://github.com/datafuselabs/databend/blob/609e02bf70680e319b0ebedf3c19f63ac3f85d8d/src/query/sql/src/planner/binder/project.rs#L263-L276
ok， let me try it.

**Review by andylokandy — APPROVED** [review]

**Comment by JackTan25** [comment]
thanks @andylokandy 's help.

File: src/query/sql/src/planner/binder/project.rs [hunk 2]
```diff
@@ -260,10 +270,16 @@ impl Binder {
                     scalar_binder.allow_pushdown();
                     let (bound_expr, _) = scalar_binder.bind(expr).await?;
 
+                    let mut expr = expr.clone();
+                    let mut remove_quote_visitor = RemoveIdentifierQuote;
```
Comment (andylokandy) [inline src/query/sql/src/planner/binder/project.rs:274]: why not move these?

File: src/query/sql/src/planner/binder/project.rs [hunk 3]
```diff
@@ -260,10 +270,16 @@ impl Binder {
...
+
                     // If alias is not specified, we will generate a name for the scalar expression.
                     let expr_name = match alias {
                         Some(alias) => normalize_identifier(alias, &self.name_resolution_ctx).name,
-                        None => format!("{:#}", expr).to_lowercase(),
+                        None => {
+                            walk_expr_mut(&mut remove_quote_visitor, &mut expr);
+                            format!("{:#}", expr).to_lowercase()
```
Comment (andylokandy) [inline src/query/sql/src/planner/binder/project.rs:281]:
```suggestion
                            let mut expr = expr.clone();
                            walk_expr_mut(&mut RemoveIdentifierQuote, &mut expr);
                            format!("{:#}", expr).to_lowercase()
```

**Review by b41sh — APPROVED** [review]

**Review by youngsofun — APPROVED** [review]

**Comment by youngsofun** [comment]
the only error is, let us merge it

```
-   HashJoin: INNER (rows: 30033.74)
+   HashJoin: INNER (rows: 30031.34)
```

**Pull request merged**